spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -191,13 +200,17 @@ private void deployStream(StreamDefinition stream, Map<String, String> streamDep
 
 			currentModule = postProcessLibraryProperties(currentModule);
 
-			this.deployer.deploy(new ModuleDeploymentRequest(currentModule, coordinates, moduleDeploymentProperties));
+			AppDefinition definition = new AppDefinition(currentModule.getLabel(), currentModule.getParameters());
+			MavenResource resource = MavenResource.parse(coordinates.toString(), mavenProperties);
+			AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, moduleDeploymentProperties);
+			this.deployer.deploy(request);
```

Comment:
I know this is still broken and can be done later but just a note, id from deploy needs to be stored somewhere so that it can be use with undeploy()/status().